src/components/body/SearchResult.tsx

```diff
@@ -9,20 +10,21 @@ export interface SearchResultProps {
 }
 
 function SearchResult({ searchTerm, columnsCount }: SearchResultProps) {
-	const [ searchResult, setSearchResult ] = useState<TenorResult>(null!);
-	const [ isLoading, setLoading ] = useState(true);
+	const [searchResult, setSearchResult] = useState<TenorResult>(null!);
+	const [isLoading, setLoading] = useState(true);
+	const [pickerContext] = useContext(PickerContext);
 	const tenor = useContext(TenorContext);
 
 	useEffect(() => {
 		setLoading(true);
 		async function search(): Promise<any> {
-			const result = await tenor.search(searchTerm);
+			const result = await tenor.search(searchTerm || pickerContext.initialSearchTerm);
```

Comment:
Initially I thought about adding the logic to the `TenorManager` but since this is the actual place that triggers the search it was looking better to leave this logic here.